ReactSkia/components/RSkComponentTextInput.cpp

```diff
@@ -6,93 +6,154 @@
-  *stopPropagation=false;
+ * @brief       Handling the key Event in TextInput.
+ * @param       eventKeyType, is Key type
+ * @param[out]  StopPropogation set false if evnt should be progated furtuer
+ * (bubbling) else true if you want to stop propogation of event
+ * @return      True if key is handled else false
+ */
+
+void RSkComponentTextInput::onHandleKey(rnsKey eventKeyType, bool *stopPropagation) {
+  *stopPropagation = false;
   RNS_LOG_TODO("update Keymatrics Event Count.");
-  //TODO  update Keymatrics Event Count.
+  // TODO  update Keymatrics Event Count.
   KeyPressMetrics keyPressMetrics;
   TextInputMetrics textInputMetrics;
   auto component = getComponentData();
   auto textInputEventEmitter = std::static_pointer_cast<TextInputEventEmitter const>(component.eventEmitter);
-  if (eventKeyType == RNS_KEY_Select){
-    if(isInEditingMode_ == false){
+  if (eventKeyType == RNS_KEY_Select) {
+    if (isInEditingMode_ == false) {
       RNS_LOG_TODO("onfocus need to here");
       /*TODO Check onfocus need to here.
-      *textInputEventEmitter->onfocus();
+      textInputEventEmitter->onfocus();
       */
-      isInEditingMode_=true;
-    }else{
+      isInEditingMode_ = true;
+    } else {
       /* TODO Update textinputmatrix data and sent the event.
       textInputEventEmitter->onSubmitEditing(textInputMetrics_);
       textInputEventEmitter->onBlur(textInputMetrics_);
       textInputEventEmitter->onEndEditing(textInputMetrics_);
       */
-      isInEditingMode_=false;
+      isInEditingMode_ = false;
       *stopPropagation = true;
     }
-  }else{
-    if(isInEditingMode_){
-      if ((eventKeyType >= RNS_KEY_1 && eventKeyType <= RNS_KEY_z)){
-        keyPressMetrics.text=RNSKeyMap[eventKeyType]; 
+  } else {
+    if (isInEditingMode_) {
+      if ((eventKeyType >= RNS_KEY_1 && eventKeyType <= RNS_KEY_z)) {
+        keyPressMetrics.text = RNSKeyMap[eventKeyType];
         textInputEventEmitter->onKeyPress(keyPressMetrics);
         RNS_LOG_TODO("send onchange and onchangetext here");
         /*TODO
-        *send onchange and onchangetext here.
+        send onchange and onchangetext here.
         */
+
+
         *stopPropagation = true;
       }
     }
   }
 }
 RnsShell::LayerInvalidateMask  RSkComponentTextInput::updateComponentProps(const ShadowView &newShadowView,bool forceUpadate){
-  // TODO
+  auto const &textInputProps = *std::static_pointer_cast<TextInputProps const>(newShadowView.props);
+
+  if (textInputProps.value.size()) {
+    displayString_ = textInputProps.value.c_str();
+    textString_ = displayString_;
+  }else if (textInputProps.defaultValue.size()) {
+    displayString_ = textInputProps.defaultValue.c_str();
+    textString_ = displayString_;
+  }else {
+    displayString_ = textInputProps.placeholder.c_str();
+    placeholderColor_ = textInputProps.placeholderTextColor;
+  }
+
   return RnsShell::LayerInvalidateNone;
```

Comment:
we should add a check if value/default value/placeholder has changed,then return should be LayerPaintValidate else None